src/transformState.js

```
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        };
+
+        break;
+
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const keyRemove of action.keysToRemove) {
+          delete state[keyRemove];
+        };
+
+        break;
+
+      default:
+        return 'Error';
+    }
```

Let's throw an error here instead.
Read more - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw